container/container_unix.go

```diff
@@ -343,6 +343,17 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network) ([]
 		createOptions = append(createOptions, libnetwork.EndpointOptionGeneric(genericOption))
 	}
 
+	/* FIXME: When restoring, the container should get the same IP address it had before, so
```

Comment:
If it is going to be a best effort and ok with failure for already allocated IP, then the code to specify the preferred address in the endpoint ipam options would probably go in here, but modified. Anyways, the required API are not available yet to be called from docker/docker, this cannot be taken care now.